Fix images links

jamesetouma opened the pull request:
Fixed image links for Python_Tutorials

oskooi commented:
Could you explain why this change is necessary?

jamesetouma commented:
The images were not rendered for me on Firefox or Brave. Do they render for you if you view the markdown files in the Python_Tutorial folder? I believe it has to do with <center>...</center>. I'm not sure if gitlab changed their markdown API.

oskooi commented:
I think this is a necessary change as well because [`<center>`](https://www.w3schools.com/tags/tag_center.asp) is an HTML tag used for aligning text and not images. (It was set up this way by @HomerReid when he created the first version of the documentation using markdown.) Using `<center>` for images works on Chrome and the reason it has not been an issue is because we have not really tested it on other browsers until now.

oskooi commented on doc/docs/Python_Tutorials/Basics.md:
```diff
@@ -51,7 +51,9 @@ geometry = [mp.Block(mp.Vector3(mp.inf,1,mp.inf),
 
-<center>![](../images/Python-Tutorial-wvg-straight-eps-000000.00.png)</center>
+<p align="center">
+  <img  src="../images/Python-Tutorial-wvg-straight-eps-000000.00.png">
```
Remove the extra space between `img` and `src`. This also applies to several other instances in this PR.

oskooi commented on doc/docs/Python_Tutorials/Basics.md:
```diff
@@ -1115,12 +1126,12 @@ unix% convert ring-ez-*.png ring-ez-0.118.gif
 
 The resulting animations for (from left to right) 0.118, 0.147, and 0.175, are below, in which you can clearly see the radiating fields that produce the losses:
+<p align="center">
+  <img src="../images/Tut-ring-ez-0.118.gif">
```
Remove extra blank space to correct alignment.

oskooi commented on doc/docs/Python_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity.md:
```diff
@@ -408,10 +408,18 @@ It is usually a good idea to examine the field patterns for any modes that you a
 + $k_x=0.25$, $ω=0.2506$ light-cone (extended) mode
+<p align="center">
+  <img  src="../images/Holey-wvg-kx=0.4-w=0.1896.gif">
+
```
Remove blank lines between `img` lines.

jamesetouma commented:
changes have been made

oskooi commented on doc/docs/Python_Tutorials/Cylindrical_Coordinates.md:
```diff
@@ -163,11 +167,9 @@ For a given mode of the ring resonator, it is often useful to know how sensitive
 
 [Pertubation theory for Maxwell equations involving high index-contrast dielectric interfaces](http://math.mit.edu/~stevenj/papers/KottkeFa08.pdf) is reviewed in Chapter 2 of [Photonics Crystals: Molding the Flow of Light, 2nd Edition (2008)](http://ab-initio.mit.edu/book/). The formula (equation 30 on p.19) for the frequency shift $\Delta \omega$ resulting from the displacement of a block of $\v ...
 
-<center>
```
Is it possible to also apply the `<p align="center">...</p>" HTML construct to center Latex equations?

jamesetouma replied:
I'll give it a try later today.

jamesetouma replied:
These three gave the same display. 
The \$\$ ... \$\$ is supposed to center it, at least from the pure $\LaTeX$ behavior. 


<center>

$$ \sigma_{scatt} = \int_0^{2\pi} d\phi \int_0^{\pi} \sigma_{diff}(\phi,\theta)\sin(\theta)d\theta $$

</center>

$$ \sigma_{scatt} = \int_0^{2\pi} d\phi \int_0^{\pi} \sigma_{diff}(\phi,\theta)\sin(\theta)d\theta $$

<p align="center">
   $$ \sigma_{scatt} = \int_0^{2\pi} d\phi \int_0^{\pi} \sigma_{diff}(\phi,\theta)\sin(\theta)d\theta $$
</p>

oskooi commented on doc/docs/Python_Tutorials/Optical_Forces.md:
```diff
@@ -3,10 +3,10 @@
 This tutorial demonstrates Meep's ability to compute classical forces via the [Maxwell stress tensor](https://en.wikipedia.org/wiki/Maxwell_stress_tensor). The geometry consists of two identical, parallel, silicon waveguides with square cross section in vacuum. A schematic of the geometry is shown below. Due to the parallel orientation of the waveguides (propagation axis along $z$ and separation  ...
+<p align="center">
+  <img src="../images/Waveguide_forces.png">
+    </p>
```
`</p>` is indented incorrectly.

jamesetouma replied:
I'll go through one more time, fix the indentations, and remove the "center" from the equations (and just keep \$\$ \... \$\$)

oskooi commented:
Note that it is not necessary to create a separate commit for each change to each file. You can (and in this case should) combine all the changes within a single commit.

jamesetouma commented:
I was editing in the browser. I'll keep that in mind in the future. Maybe you can squash those commits if these edits get accepted.

oskooi reviewed (approved):
LGTM.

stevengj commented:
Thanks!

oskooi commented:
Unfortunately, this PR has removed all the images from the online user manual, e.g., see:
https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/

jamesetouma commented:
Yes I saw that. The html tags maintain their relative path; ie ../images/x.png, while the site folder needs ../../images/x.png to render correctly. I am not sure why this is the case. However, if we use plain markdown tags ( ![](image name) ), they both render correctly, however, the images will not be centered.

jamesetouma commented:
I submitted another PR -  plz see comment in the request.

oskooi commented:
One solution to centering images in markdown *without* adding HTML tags is via CSS: https://msudol.com/how-to-align-images-in-markdown-without-html-or-extensions/. We can do something like:
```
![<>](../image.png)
```
(i.e., adding the `<>` inside the square brackets.)

Perhaps we can test this feature on just a single image first via a PR and if it works apply it to all the other images (via another PR).

jamesetouma commented:
Sure!

jamesetouma commented:
Just updated my PR branch. 

css/extra.css is updated with the two methods
Python_tutorials/Basics.md: updated the last image in the file using both methods. Please let me know which one you prefer. 

I noticed that the MD file did not change but the HTML file did. Maybe you want to check it on your side.

oskooi commented on doc/docs/Python_Tutorials/Local_Density_of_States.md:
```diff
@@ -222,7 +223,7 @@ if __name__ == '__main__':
 -------------------------------
 
-We consider the simple example of a 2D perfect-metal $a$x$a$ cavity of finite thickness 0.1$a$, with a small notch of width $w$ on one side that allows the modes to escape. The nice thing about this example is that in the absence of the notch, the lowest-frequency $E_z$-polarized mode is known analytically to be $E_z^{(1)}=\frac{4}{a^2}\sin(\pi x/a)\sin(\pi \gamma/a)$, with a frequency $\omega^{( ...
+We consider the simple example of a 2D perfect-metal $a$x$a$ cavity of finite thickness 0.1$a$, with a small notch of width $w$ on one side that allows the modes to escape. The nice thing about this example is that in the absence of the notch, the lowest-frequency $E_z$-polarized mode is known analytically to be $E_z^{(1)}=\frac{4}{a^2}\sin(\pi x/a)\sin(\pi \gamma/a)$, with a frequency $\omega^{( ...
```
Putting *two* slashes before the `#` character in the equation on the last line of this paragraph is causing it not to render in the online user manual: https://meep.readthedocs.io/en/latest/Python_Tutorials/Local_Density_of_States/ .

jamesetouma replied:
Typo - I'll fix
